dbtmetabase/parsers/dbt_folder.py

```diff
@@ -0,0 +1,196 @@
+import re
+import os
+import logging
+
+from pathlib import Path
+from typing import List
+
+import yaml
+
+from dbtmetabase.models.metabase import METABASE_META_FIELDS
+from dbtmetabase.models.metabase import MetabaseModel, MetabaseColumn
+
+
+class DbtFolderReader:
+    """
+    Reader for dbt project configuration.
+    """
+
+    def __init__(self, project_path: str):
+        """Constructor.
+
+        Arguments:
+            project_path {str} -- Path to dbt project root.
+        """
+
+        self.project_path = (
+            project_path
+            if project_path[0] != "~"
+            else (os.path.expanduser("~") + project_path[1:])
+        )
+        self.catch_aliases = {}
+
+    def read_models(
+        self,
+        database: str,
+        schema: str,
+        schemas_excludes: list = [],
```

Comment:
Not a big deal, but I think it's a best practice that type hinting on arguments should be as permissive as possible. For example: Instead of requiring consumers to pass a list, I think any `Iterable` would do here. https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#standard-duck-types. But then, return types should be as specific as possible.

I think we are also getting a bit inconsistent here using `list` and `List` as type hints. If I'm not mistaken, we can use `list` everywhere as long as we don't mind making this Python 3.9+ only. https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#standard-duck-types

Reply:
Updated type hints to be compatible with pre 3.9 releases of python. Made them all consistent too. Thanks for the link, I'll be putting that principle into practice more in my own coding- thanks!